ResumableUpload.CS.Sample/ResumeableUploadSample.cs

```diff
@@ -0,0 +1,241 @@
+                                        {Location = new GeoPoint 
+                                            {Latitude = 41.327464, Longitude = -72.194555, Altitude = 10},
+                                            LocationDescription = "East Lyme, CT, U.S.A.",
+                                            RecordingDate=DateTime.Now.Date};
+                    string strFilePath = "C:\\Users\\YourUserName\\Videos\\Video\\YourVideo.mp4";
+                    Globals.FileLength = new FileInfo(strFilePath).Length;
+                    Globals.ProgressPercent = 0;
+                    using (FileStream objFileStream = new FileStream(strFilePath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+                        {
+                        objVideosInsertRequest = objYouTubeService.Videos.Insert(objVideo, "snippet,status,recordingDetails", objFileStream, "video/*");
+                        objVideosInsertRequest.ChunkSize = intHalfMBChunkSize;  // .5MB
+                        objVideosInsertRequest.ResponseReceived += videosInsertRequest_ResponseReceived; // Add event handler
+                        objVideosInsertRequest.ProgressChanged += videosInsertRequest_ProgressChanged;   // Add event handler
+                        objCancellationToken = new System.Threading.CancellationToken();
+                        Globals.ResumeRetries = 0;
+                        try
+                        {
+                            await objVideosInsertRequest.UploadAsync(objCancellationToken);
+                        }
+                        catch
+                        {
+                            // Error handling is done in videosInsertRequest_ProgressChanged() Event
+                        }
+                        bool bResumeAsyncWasLaunched = false;
+                        do
+                        {
+                            Thread.Sleep(3000); // Allow time for ProgressChanged() Event to execute and to allow time for a server or network issue to resolve
+                            bResumeAsyncWasLaunched = false;
+                            if (IsResumeable())
+                            {
+                                // Try a ResumeAsync
+                                Globals.ResumeRetries += 1;
+                                bResumeAsyncWasLaunched = true;
+                                try
+                                {
+                                   await objVideosInsertRequest.ResumeAsync(objCancellationToken);
+                                }
+                                catch
+                                {
+                                    // Error handling is done in videosInsertRequest_ProgressChanged() Event
+                                }
+                            }
+                        } while (bResumeAsyncWasLaunched && IsResumeable());
+                        objFileStream.Close();
+                    }
+                }
+            }
+        }
```

Comment:
This function is pretty long. Would it be possible to split it into several smaller functions?